src/csharp/Grpc.Core/Internal/PlatformInvocation.cs

```diff
@@ -0,0 +1,257 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Grpc.Core.Internal
+{
+    /// <summary>
+    /// Platform invocation dispatch
+    /// </summary>
+    internal static class PlatformInvocation
+    {
+        /// <summary>
+        /// Retrieve an implementation of platform invocations specific to the current cpu's architecture.
+        /// This allows assemblies to function correctly when compiled for AnyCPU
+        /// </summary>
+        internal static IPlatformInvocation Implementation { get; private set; }
```

Comment:
I understand the idea, but can you please summarize what exactly are the differences in different implementations of IPlatformInvocations?  It seems like 99% of the stuff is identical. Can we focus on looking at the differences and figuring out what we can do to prevent the need to have the same definition copy and pasted 3 times?